GoogleSignIn/Tests/Unit/GIDDataFetcherTest.m

```diff
@@ -0,0 +1,135 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GoogleSignIn/Sources/GIDDataFetcher/Implementations/GIDDataFetcher.h"
+
+#import <XCTest/XCTest.h>
+
+#import "GoogleSignIn/Tests/Unit/GIDFakeFetcher.h"
+#import "GoogleSignIn/Tests/Unit/GIDFakeFetcherService.h"
+#import "GoogleSignIn/Tests/Unit/OIDAuthState+Testing.h"
+
+#ifdef SWIFT_PACKAGE
+@import AppAuth;
+@import GTMAppAuth;
+@import OCMock;
+#else
+#import <AppAuth/AppAuth.h>
+#import <GTMAppAuth/GTMAppAuth.h>
+#import <OCMock/OCMock.h>
+#endif
+
+static NSString *const kTestURL = @"testURL.com";
+static NSString * const kErrorDomain = @"ERROR_DOMAIN";
+static NSInteger const kErrorCode = 212;
+
+@interface GIDDataFetcherTest : XCTestCase {
+  GIDDataFetcher *_dataFetcher;
+  
+  // Mock |GTMAppAuthFetcherAuthorization|.
+  id _authorization;
+  
+  // Fake fetcher service to emulate network requests.
+  GIDFakeFetcherService *_fakeFetcherService;
+}
+
+@end
+
+@implementation GIDDataFetcherTest
+
+- (void)setUp {
+  [super setUp];
+  _dataFetcher = [[GIDDataFetcher alloc] init];
+  _fakeFetcherService = [[GIDFakeFetcherService alloc] init];
+  
+  _authorization = OCMStrictClassMock([GTMAppAuthFetcherAuthorization class]);
+  OCMStub([_authorization alloc]).andReturn(_authorization);
+  OCMStub([_authorization initWithAuthState:OCMOCK_ANY]).andReturn(_authorization);
+  OCMStub([_authorization fetcherService]).andReturn(_fakeFetcherService);
+}
+
+- (void)tearDown {
+  _authorization = nil;
+  [super tearDown];
+}
+
+- (void)testFetchData_success {
+  NSURL *url = [NSURL URLWithString:kTestURL];
+  OIDAuthState *authState = [OIDAuthState testInstance];
+  XCTestExpectation *expectation =
+      [self expectationWithDescription:@"Callback called with no error"];
+
+  [_dataFetcher startFetchURL:url
+                fromAuthState:authState
+                  withComment:@"Test data fetcher."
+                   completion:^(NSData *data, NSError *error) {
+    if (error == nil) {
+      [expectation fulfill];
+    }
```

Comment:
If the completion is invoked with a non-nil error, the expectation will not fulfill and the test will time out, which is not ideal.

A better way to do this is to use the expectation to make sure the callback is invoked, and use assert to make sure the error is nil.